src/dvc_data/checkout.py

```diff
@@ -203,32 +202,35 @@ def _checkout(
         )
 
     failed = []
-    for change in chain(diff.added, diff.modified):
-        entry_path = (
-            fs.path.join(path, *change.new.key)
-            if change.new.key != ROOT
-            else path
-        )
-        if change.new.oid.isdir:
-            fs.makedirs(entry_path)
-            continue
 
-        try:
-            _checkout_file(
-                link,
-                entry_path,
-                fs,
-                change,
-                cache,
-                force,
-                relink,
-                state=state,
-                prompt=prompt,
+    bulk_save = state.bulk_save(fs) if state else nullcontext(noop)
+    with bulk_save as save:
```

Comment:
This seems kinda ugly. We should probably just load it and dump it in the data index instead.